### PR TITLE
Man in the middle

### PR DESCRIPTION
**Pull request by appasalvi**
Hi Michael and Aaron,

There are some issues with this commit (e.g. we get 4 extra tabs when we execute this code).
 The icons are working fine (i.e. they turn red when we go to websites like serebii.net).

Sorry I didn't commit earlier but I wanted to make some more improvements first to the requestinterceptor and manifest files.

Thanks.
Satya

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -13,7 +13,11 @@
 	"permissions": [
 		"contextMenus",
 		"tabs",
-		"storage"
+		"storage",
+		"scripting"
+	],
+	"host_permissions":[
+		"https://www.serebii.net/"
```
Comment (TheENGR) [inline src/manifest.json]: Probs need more

Comment (TheENGR) [reply src/manifest.json]:
```suggestion
		"https://*/"
```

File: src/manifest.chrome.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Cyber Citizenship Coefficient",
+  "description": "Build an Chrome Plugin Extension",
```
Comment (TheENGR) [inline src/manifest.chrome.json]: Do we need this at all?

File: src/siteQualityMapping.js [hunk 3]
```diff
@@ -46,10 +46,12 @@ chrome.contextMenus.onClicked.addListener(async (info, tab) => {
 
 
 chrome.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
+	console.log('onUpdate');
```
Comment (TheENGR) [inline src/siteQualityMapping.js]: Don’t push debugging logs

File: src/siteQualityMapping.js [hunk 4]
```diff
@@ -46,10 +46,12 @@ chrome.contextMenus.onClicked.addListener(async (info, tab) => {
...
 	await checkURL(tab.url.split('/')[2])
 })
 
 chrome.tabs.onActivated.addListener(async (activeInfo) => {
+	console.log('onActivated');
```
Comment (TheENGR) [inline src/siteQualityMapping.js:79]: Remove debugging logs

File: src/requestinterceptor.js [hunk 5]
```diff
@@ -9,4 +9,22 @@ chrome.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
     console.log(tabId)
     console.log(changeInfo)
     console.log(tab)
+    const URL = tab.url.split('/')[2]
+    await chrome.storage.local.get(["blockedSites", "warnedSites"], async (storage) => {
+        if (storage.warnedSites.find(x => URL.includes(x)) || storage.blockedSites.find(x => URL.includes(x))) {
+            // close target
+            chrome.tabs.remove(tabId)
```
Comment (TheENGR) [inline src/requestinterceptor.js:39]: Why not just update the tab to have a new URL?

https://developer.chrome.com/docs/extensions/reference/tabs/#method-update

**Review by TheENGR — CHANGES_REQUESTED** [review]

**Comment by TheENGR** [comment]
Definitely solid forward progress, but let’s not merge it just yet :)

File: src/manifest.chrome.json (hunk 2 above)
Comment (appasalvi) [reply src/manifest.chrome.json]: Hi TheENGR,

I deleted the manifest.chrome.json file.  

Thanks.
Satya

File: src/siteQualityMapping.js (hunk 3 above)
Comment (appasalvi) [reply src/siteQualityMapping.js]: Hi TheENGR,

This has been resolved as I have removed the console.log lines.

File: src/manifest.chrome.json (hunk 2 above)
Comment (TheENGR) [reply src/manifest.chrome.json]: You need to push that deletion

File: src/requestinterceptor.js (hunk 5 above)
Comment (appasalvi) [reply src/requestinterceptor.js:39]: Hi TheENGR,

What is the advantage of not uploading a new tab?  If the "Man in the Middle" is created within the existing tab then we could run into issues with the browser history.

Comment (TheENGR) [reply src/requestinterceptor.js:39]: We MIGHT be able to re-write the history

Wanted to not load jerks so they don't get paid for ADs

Comment (TheENGR) [reply src/requestinterceptor.js:39]: https://developer.chrome.com/docs/extensions/reference/history/ 

We can re-write the history if needed

File: src/contentScript.js [hunk 6]
```diff
@@ -1,13 +1,14 @@
 // Listen for right clicks then send data on what was clicked
 document.addEventListener('contextmenu', e => {
 	// Send messages to ourself
-	chrome.runtime.sendMessage("afffkcmpebnikjnoagiiofainbpffnch", {
+	chrome.runtime.sendMessage("GoingBack", {
```
Comment (TheENGR) [inline src/contentScript.js:4]: That string should have to be the app code, as it was

File: src/display.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    });
+    // window.location = findParam('url')+"?skip_interceptor" //setting flag so that the interceptor ignores this request
+}
+
+function alwaysGo()
+{
+    chrome.runtime.sendMessage("afffkcmpebnikjnoagiiofainbpffnch",
+    {
+        action: "always go",
+        url: findParam('url')+"?skip_interceptor" //setting flag so that the interceptor ignores this request
```
Comment (TheENGR) [inline src/display.js:34]: You can't just add a `?`, there probably is one on most sites already 
https://www.wsj.com/articles/miami-condo-buildings-old-age-collapse-champlain-towers-11626271508?mod=hp_lead_pos5
See

**Review by TheENGR — APPROVED** [review]
Solid progress! 

Time for a handoff 🔋

**Pull request merged**